relayer/metrics/metricsService.go

```diff
@@ -0,0 +1,175 @@
+	intervalData := os.Getenv("METRICS_INTERVAL_FOR_SENDING_DATA_INM")
+	if metricPortalSqsUrl == "" || intervalData == "" {
+		return nil
+	}
+	intervalForMetrics, _ := strconv.ParseInt(intervalData, 10, 32)
+	utils.LavaFormatDebug("env variables:", &map[string]string{
+		"url":      metricPortalSqsUrl,
+		"interval": intervalData})
+	mChannel := make(chan RelayAnalytics)
+	result := &MetricService{
+		metricsChannel:      mChannel,
+		sqsUrl:              metricPortalSqsUrl,
+		aggregatedMetricMap: &map[string]map[string]map[string]*AggregatedMetric{},
+	}
+	//setup reader
+	go result.ReadFromChannel()
+	//setup sending of the results into the query
+	ticker := time.NewTicker(time.Duration(intervalForMetrics * time.Minute.Nanoseconds()))
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				{
+					data := result.ConvertDataToArray()
+					err := result.SendTo(data)
+					if err != nil {
+						utils.LavaFormatError("error sending metrics data portal app.", err, nil)
+					}
+				}
+			}
+		}
+	}()
+	return result
+}
+
+func (m *MetricService) SendTo(data []RelayAnalyticsDTO) error {
+	if data == nil || len(data) == 0 {
+		return nil
+	}
+	jsonValue, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+	resp, err := http.Post(m.sqsUrl, "application/json", bytes.NewBuffer(jsonValue))
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("couldn't send the data into the server")
+	}
+	return nil
+}
+
+func (m *MetricService) ConvertDataToArray() []RelayAnalyticsDTO {
+	if m.numberOfRelays == 0 || m.aggregatedMetricMap == nil {
+		return nil
+	}
+	toSendData := make([]RelayAnalyticsDTO, m.numberOfRelays)
+	counter := 0
+	//project-chain-apiType
+	for projectKey, projectData := range *m.aggregatedMetricMap {
+		for chainKey, chainData := range projectData {
+			for apiTypekey, apiTypeData := range chainData {
+				toSendData[counter] = RelayAnalyticsDTO{
+					ProjectHash:  projectKey,
+					APIType:      apiTypekey,
+					ChainID:      chainKey,
+					Latency:      apiTypeData.TotalLatency,
+					ComputeUnits: apiTypeData.RelaysCount,
+					SuccessCount: apiTypeData.SuccessCount,
+				}
+				counter++
+			}
+		}
+	}
+	m.aggregatedMetricMap = &map[string]map[string]map[string]*AggregatedMetric{}
+	m.numberOfRelays = 0
+	return toSendData
+}
+
+func (m *MetricService) CloseChannel() {
+	if m.metricsChannel != nil {
+		m.CloseChannel()
+	}
+}
+
+func (m *MetricService) SendDataToChannel(data RelayAnalytics) {
+	if m.metricsChannel != nil {
+		m.metricsChannel <- data
+	}
+}
+
+func (m *MetricService) ReadFromChannel() {
+	if m.metricsChannel != nil {
+		for {
+			data, newMessageInserted := <-m.metricsChannel
+			if !newMessageInserted {
+				continue
+			}
+			toInsertData := &AggregatedMetric{
+				TotalLatency: data.Latency,
+				RelaysCount:  data.ComputeUnits,
+				SuccessCount: 0,
+			}
+			if data.Success {
+				toInsertData.SuccessCount = 1
+			}
```

Comment:
why are you convert to this object if it's the same? hust use the RelayAnalytics as the channel source.

I started to wrote the AggregatedMetric for the final aggregation (Totals of many metrics together)

Reply:
this is not the same. here we store the count and success count which is different from what we are getting